lib/src/web_socket_handler.dart

```diff
@@ -10,16 +10,16 @@ import 'package:web_socket_channel/web_socket_channel.dart';
 /// A class that exposes a handler for upgrading WebSocket requests.
 class WebSocketHandler {
   /// The function to call when a request is upgraded.
-  final Function _onConnection;
+  final Function? _onConnection;
```

Comment:
It looks like this was not intended to ever accept null to me

Reply:
sure